diff --git a/core/vm/jump_table.go b/core/vm/jump_table.go
index 82c5bc4e2a..82fc43ec13 100644
--- a/core/vm/jump_table.go
+++ b/core/vm/jump_table.go
@@ -123,7 +123,7 @@ func newCancunInstructionSet() JumpTable {
 }
 
 func newShanghaiInstructionSet() JumpTable {
-	instructionSet := newLondonInstructionSet()
+	instructionSet := newMergeInstructionSet()
 	enable3855(&instructionSet) // PUSH0 instruction
 	enable3860(&instructionSet) // Limit and meter initcode
 
